velox/exec/ExchangeQueue.cpp

```diff
@@ -86,7 +96,13 @@ void ExchangeQueue::enqueueLocked(
   receivedBytes_ += page->size();
 
   queue_.push_back(std::move(page));
-  if (!promises_.empty()) {
+  const auto minBatchSize = minOutputBatchBytesLocked();
+  while (!promises_.empty()) {
+    const auto unblockedConsumers = numberOfConsumers_ - promises_.size();
```

Comment:
```
VELOX_CHECK_GE(numberOfConsumers_, promises_.size());
```
We probably only able to add this check after removing the backward compatibility check.

Reply:
I think it should hold even now. Let me try.